microsoft/testsuites/mdatp/mdatp_basic.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import time
+from typing import Any
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import BSD
+from lisa.testsuite import TestResult
+from lisa.tools import MDE, Curl
+from lisa.util import LisaException, SkippedException
+
+from microsoft.testsuites.mdatp.mdatp_tools import Mdatp
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="""
+        Verify MDE installation
+        Microsoft Defender for Endpoint(MDE) for Linux includes
+        antimalware and endpoint detection and response (EDR) capabilities.
+
+        This test suites validates if MDE can be installed, onboarded
+        and detect an EICAR file.
+
+        The test requires the onboarding script to be kept in Azure Storage Account
+        and provide the SAS url for downloading under the
+        secret variable `onboarding_script_sas_uri`.
+
+        The suite runs the following tests:
+        1. Installation test
+        2. Onboarding test
+        3. Health test
+        4. EICAR detection test
+    """,
+)
+class MDETest(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        variables = kwargs["variables"]
+        self.onboarding_script_sas_uri = variables.get("onboarding_script_sas_uri", "")
+        if not self.onboarding_script_sas_uri:
+            raise SkippedException("Onboarding script SAS URI is not provided.")
+
+    @TestCaseMetadata(
+        description="""
+            Verify MDE installation, onboarding, health and EICAR detection.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            min_core_count=2, min_memory_mb=1024, unsupported_os=[BSD]
+        ),
+    )
+    def verify_mde(self, node: Node, log: Logger, result: TestResult) -> None:
+        # Invoking tools first time, intalls the tool.
+        try:
+            output = node.tools[Mdatp]._check_exists()
```

Comment:
1. Don't call the private method like `_check_exists`, call `exists`.
2. But when you refer a tool like this, the check/install happens automatically. So, the code can be like `_ = node.tools[Mdatp]`, and capture the exception if there is.